src/test/java/de/retest/recheck/util/ReflectionUtilitiesTest.java

```diff
@@ -0,0 +1,250 @@
+		Mockito.when( e3.getStackTrace() ).thenReturn( value3 );
+		Mockito.when( e3.getCause() ).thenReturn( e2 );
+
+		assertThat( ReflectionUtilities.isThreadDeathWhileClosingSuT( e3 ) ).isFalse();
+	}
+
+	private static ThreadDeath getThreadDeathWhileClosingSuTEx() {
+		final ThreadDeath e = Mockito.mock( ThreadDeath.class );
+		@SuppressWarnings( "restriction" )
+		final StackTraceElement[] value =
+				{ new StackTraceElement( sun.awt.AppContext.class.getCanonicalName(), "dispose", "", 0 ) };
+		Mockito.when( e.getStackTrace() ).thenReturn( value );
+		return e;
+	}
+
+	@Test
+	public void instanceOf_Class_String_should_handle_equal_types() throws Exception {
+		final Class<?> instanceClass = A.class;
+		final String className = instanceClass.getName();
+
+		final boolean actual = ReflectionUtilities.instanceOf( instanceClass, className );
+
+		assertThat( actual ).isTrue();
+	}
+
+	@Test
+	public void instanceOf_Class_String_should_handle_nested_types() throws Exception {
+		final Class<?> instanceClass = B.class;
+		final String className = instanceClass.getName();
+
+		final boolean actual = ReflectionUtilities.instanceOf( instanceClass, className );
+
+		assertThat( actual ).isTrue();
+	}
+
+	@Test
+	public void instanceOf_Class_String_should_handle_multiple_nested_types() throws Exception {
+		final Class<?> instanceClass = C.class;
+		final String className = instanceClass.getName();
+
+		final boolean actual = ReflectionUtilities.instanceOf( instanceClass, className );
+
+		assertThat( actual ).isTrue();
+	}
+
+	@Test
+	public void instanceOf_Class_String_should_handle_unrelated_types() throws Exception {
+		final Class<?> instanceClass = A.class;
+		final String className = B.class.getName();
+
+		final boolean actual = ReflectionUtilities.instanceOf( instanceClass, className );
+
+		assertThat( actual ).isFalse();
+	}
+
+	@Test
+	public void instanceOf_Class_String_should_handle_usage_of_toString() throws Exception {
+		final Class<?> instanceClass = A.class;
+		final String className = instanceClass.toString();
+
+		expectedException.expect( IllegalArgumentException.class );
+		expectedException.expectMessage( "Class name starts with 'class '. This is probably due to the use of "
+				+ "Object#toString(), whereas Class#getName() should be used." );
+		ReflectionUtilities.instanceOf( instanceClass, className );
+	}
+
+	@Test
+	public void instanceOf_String_Class_should_handle_existent_types() throws Exception {
+		final Class<?> clazz = A.class;
+		final String instanceClassName = clazz.getName();
+
+		final boolean actual = ReflectionUtilities.instanceOf( instanceClassName, clazz );
+
+		assertThat( actual ).isTrue();
+	}
+
+	@Test
+	public void instanceOf_String_Class_should_handle_inexistent_types() throws Exception {
+		final Class<?> clazz = A.class;
+		final String instanceClassName = "SomeUnkownType";
+
+		final boolean actual = ReflectionUtilities.instanceOf( instanceClassName, clazz );
```

Comment:
Nope, it fails in test setField_should_throw_meaningful_IncompatibleTypesException_on_wrong_types, and this because of
https://bugs.openjdk.java.net/browse/JDK-8217225 :man_shrugging: